src/ts/features/shortcuts.tsx

```diff
@@ -1,22 +1,110 @@
 import {Features} from './features'
-import {configure, GlobalHotKeys} from 'react-hotkeys'
+import {configure, GlobalHotKeys, KeyMap} from 'react-hotkeys'
+// @ts-ignore
+import KeyEventManager from 'react-hotkeys/es/lib/KeyEventManager'
 import React from 'react'
 import ReactDOM from 'react-dom'
+import {nativeKeyBindingsToIgnore} from './block-navigation-mode'
+import {CODE_TO_KEY, normalizeKeySequence} from '../utils/react-hotkeys'
 
 configure({
     ignoreTags: [],
     ignoreRepeatedEventsWhenKeyHeldDown: false,
+    // This emits ctrl+u when holding down ctrl and pressing u,
+    // otherwise only u is emitted
+    simulateMissingKeyPressEvents: true,
+    // https://github.com/greena13/react-hotkeys/issues/249
+    // Plain key bindings like `u` shouldn't clobber shortcuts like `cmd+u`
+    stopEventPropagationAfterHandling: false,
+    stopEventPropagationAfterIgnoring: false,
+    customKeyCodes: CODE_TO_KEY,
 })
 
 const shortcutContainer = document.createElement('div')
 
+
+type Handler = (event: KeyboardEvent) => Promise<any> | undefined
+export type Handlers = {
+    [action: string]: Handler
+}
+
+let executingHandler = 0
+
+// Roam actions such as unfocusing a block require simulating key presses.
+// We only want want that simulated "Esc" press to do it's native job, not
+// trigger another hotkey handler
+const onlyOneHandlerAtATime = (handler: Handler): Handler => async (event: KeyboardEvent) => {
+    if (executingHandler > 0) {
+        return
+    }
+
+    executingHandler += 1
+    try {
+        await handler(event)
+    } catch (error) {
+        console.error(error)
+    }
+    executingHandler -= 1
+}
+
 export async function updateShortcuts() {
+    const keyMap = await Features.getCurrentKeyMap()
+    const handlers = Features.getShortcutHandlers()
+
+    /**
+     * For some reason, key sequences like 'g g' mess up the other shortcuts
```

Comment:
hm, on the other hand it'd be to great to have sequences work, on the other hand if it requires a lot of custom code, that gives me some hesitation..